src/Common/src/Interop/Windows/mincore/Interop.QuerySystemInformation.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        internal enum SYSTEM_INFORMATION_CLASS
+        {
+            SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION = 8
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct SYSTEM_PROCESSOR_PERFORMANCE_INFORMATION
+        {
+            public long IdleTime;
+            public long KernelTime;
+            public long UserTime;
+            public long DpcTime;
+            public long InterruptTime;
+            public uint InterruptCount;
+        }
+        
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        internal unsafe delegate void QuerySystemInformationDelegate(SYSTEM_INFORMATION_CLASS SystemInformationClass, void* SystemInformation, int SystemInformationLength, out uint returnLength);
+
+        private static QuerySystemInformationDelegate s_querySystemInformation;
+
+        internal unsafe static QuerySystemInformationDelegate QuerySystemInformation
+        {
+            get
+            {
+                if(s_querySystemInformation == null)
+                {
+                    IntPtr ntDll = LoadLibraryEx("NtDll.dll", IntPtr.Zero, 0);
```

Comment:
This should pass `LOAD_LIBRARY_SEARCH_SYSTEM32` flag instead of 0 - we do not want to be loading any random ntdll.dll laying around.